File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0 , start = 1;
        boolean isEven = false;
        if(n%2==0){
            isEven = true;
            start = 2;
        }else{
            start = 1;
        }
        for(int i = start ; i<=n ; i+=2){
            if(isEven) answer+=Math.pow(i,2);
            else answer+=i;
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    let answer = 0;
    for (let i = n % 2 === 0 ? 2 : 1; i <= n; i += 2) {
        answer += n % 2 === 0 ? i * i : i;
    }    
    
    return answer;
}
